reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java

```diff
@@ -83,94 +88,112 @@ public void channelInactive(ChannelHandlerContext ctx) {
 	@Override
 	@SuppressWarnings("FutureReturnValueIgnored")
 	public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-		if (msg instanceof HttpResponse) {
-			if (((HttpResponse) msg).status().equals(HttpResponseStatus.CONTINUE)) {
-				//"FutureReturnValueIgnored" this is deliberate
-				ctx.write(msg, promise);
-				return;
-			}
-
-			dataSentTime = System.nanoTime();
-		}
+		try {
+			if (msg instanceof HttpResponse) {
+				if (((HttpResponse) msg).status().equals(HttpResponseStatus.CONTINUE)) {
+					return;
+				}
 
-		if (msg instanceof ByteBufHolder) {
-			dataSent += ((ByteBufHolder) msg).content().readableBytes();
-		}
-		else if (msg instanceof ByteBuf) {
-			dataSent += ((ByteBuf) msg).readableBytes();
-		}
+				dataSentTime = System.nanoTime();
+			}
 
-		if (msg instanceof LastHttpContent) {
-			promise.addListener(future -> {
-				ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
-				if (channelOps instanceof HttpServerOperations) {
-					HttpServerOperations ops = (HttpServerOperations) channelOps;
-					recordWrite(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path),
-							ops.method().name(), ops.status().codeAsText().toString());
-					if (!ops.isHttp2() && ops.hostAddress() != null) {
-						// This metric is not applicable for HTTP/2
-						// ops.hostAddress() == null when request decoding failed, in this case
-						// we do not report active connection, so we do not report inactive connection
-						recordInactiveConnection(ops);
+			dataSent += extractProcessedDataFromBuffer(msg);
+
+			if (msg instanceof LastHttpContent) {
+				promise.addListener(future -> {
+					try {
+						withHttpServerOperations(ctx, ops -> {
```

Comment:
Please use `try/catch` without introducing a `Consumer`

Reply:
I'll revert that change, still a reason would be much appreciated.

Reply:
Try to create JMH test and share the results from the two approaches.

Reply:
I created a very generic setup here https://github.com/ChristianLMI/lambda-bench. The repo also includes some results.

Reply:
Thanks for taking the time to setup this. It is very interesting. Can you now extend the lambda to use outer class members as it is in our case. (e.g. uriTagValue)

Reply:
I added another benchmark. Please have a look if that matches what you had in mind.

Reply:
yes